packages/dds/tree/src/feature-libraries/editable-tree/utilities.ts

```diff
@@ -142,3 +160,98 @@ export function keyIsValidIndex(key: string | number, length: number): boolean {
     if (typeof key === "string" && String(index) !== key) return false;
     return Number.isInteger(index) && 0 <= index && index < length;
 }
+
+/**
+ * Attempts to wrap an arbitrary data into a cursor, trying to resolve data types according to the given schema.
+ *
+ * Note that the name of this function as well as its return type do not specify the tree format on purpose
+ * to encourage a development of support for pluggable formats/cursors in the EditableTree API.
+ */
+export function tryGetCursorFor(
+    schema: SchemaDataAndPolicy,
+    fieldSchema: FieldSchema,
+    data: unknown,
+): ITreeCursor {
+    const node = DetachedNode.create(schema, fieldSchema, data);
+    return singleMapTreeCursor(node);
+}
+
+/**
+ * This implementation of the MapTree is written to wrap an arbitrary data with respect to the tree schema.
+ *
+ * It is used as an intermediate storage for the user data,
+ * whenever a user utilizes simple assignments to change an EditableTree,
+ * as methods of the EditableTree accept only cursors as an input data.
+ */
+export class DetachedNode implements MapTree {
```

Comment:
This could be turned into something more general, but I wanted to keep things simple.